tests/testthat/test-pmc.R

```diff
@@ -0,0 +1,103 @@
+test_that("pmc parameterisation works", {
+  p <- get_parameters()
+  
+  expect_false(p$pmc)
+  expect_equal(p$pmc_ages, NULL)
+  expect_equal(p$pmc_coverages, NULL)
+  expect_equal(p$pmc_timesteps, NULL)
+  
+  
+  p <- set_drugs(
+    parameters = p,
+    drugs = list(SP_AQ_params))
+  p <- set_pmc(
+    parameters = p,
+    drug = 1,
+    timesteps = c(5, 10),
+    coverages = c(0.5, 1),
+    ages = c(30, 60, 90)
+  )
+  
+  expect_true(p$pmc)
+  expect_equal(p$pmc_ages, c(30, 60, 90))
+  expect_equal(p$pmc_coverages, c(0.5, 1))
+  expect_equal(p$pmc_timesteps, c(5, 10))
+  expect_equal(p$pmc_drug, 1)
+})
+
+
+test_that("pmc gives drugs to correct ages", {
+
+  p <- get_parameters(list(human_population = 6))
+  p <- set_drugs(
+    parameters = p,
+    drugs = list(SP_AQ_params))
+  p$drug_efficacy <- 1
+  p <- set_pmc(
+    parameters = p,
+    drug = 1,
+    timesteps = 10,
+    coverages = 1,
+    ages = c(30, 60, 90)
+  )
+  timestep <- 10
+  events <- create_events(p)
+  renderer <- individual:::Render$new(timestep)
+  variables <- create_variables(p)
+  variables$birth <- individual::IntegerVariable$new(
+    -c(10, 30, 60, 89, 90, 36500) + 10
+  )
+  variables$state <- mock_category(
+    c('D', 'S', 'A', 'U', 'Tr'),
+    c('D', 'S', 'A', 'U', 'D', 'S')
+  )
+  variables$drug <- mock_integer(rep(0, 6))
+  variables$drug_time <- mock_integer(rep(-1, 6))
+  mockery::stub(sample_intervention, 'bernoulli', mockery::mock(c(TRUE, TRUE, TRUE)))
+  
+  process <- create_pmc_process(
+    variables = variables,
+    events = events, 
+    parameters = p,
+    renderer = renderer,
+    correlations = get_correlation_parameters(p),
+    coverages = p$pmc_coverages,
+    timesteps = p$pmc_timesteps,
+    drug = p$pmc_drug
+  )
+  
```

Comment:
Ahh, I didn't notice that it was a randomness issue. You can mock bernoulli (called from the pmc process) to make everyone's treatment successful.

This is similar to how it's done in the mda test, except there are three individuals to move instead of two.

```suggestion
  # mock the treatment success
  mockery::stub(process, 'bernoulli', mockery::mock(c(TRUE, TRUE, TRUE)))
```

Reply:
I suggest this way over `p$drug_efficacy <- 1` because it allows us to test other alternatives in the future, like people seeking treatment but it's not efficacious.